Scripts/StaminaSystem.gd

```diff
@@ -0,0 +1,43 @@
+extends Label
+
+onready var stamina_label: Label = get_node(".")
+onready var ElfStats = get_parent().get_parent().get_node("ElfStats")
+
+var current_stamina: float = 10
+var max_stamina: float = 10
+var auto_atack: bool = true
+var timer = null
+
+func _ready():
+	max_stamina = ElfStats.get_stat("stamina").value
+	current_stamina = max_stamina
+	set_process_input(true)
+
+func _input(event):
+	if event is InputEventMouseButton and event.pressed and event.button_index == 1:
+		increasing_stamina()
+
+func increasing_stamina():
```

Comment:
So many code. Isn't that better?
```gdscript
current_stamina = min(current_stamina + 1, max_stamina)
```

anyway, 'else' statement in 26 line is useless. Why we setting current stamina to the max value, if it equals that value?

Reply:
You are right